cpp/include/cudf/strings/detail/merge.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/detail/merge.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf ::strings ::detail {
+/**
+ * @brief Merges two strings columns
+ *
+ * @param lhs First column
+ * @param rhs Second column
+ * @param row_order Indices for each column
```

Comment:
I noticed that `Caller must set the validity mask in the output column.` has been removed. Is there really a change in behavior?

Reply:
Yes, the validity mask is computed in the `make_strings_column` utility and so the caller does not need to compute the mask anymore. 
It has been removed here:
https://github.com/rapidsai/cudf/pull/15374/files#diff-bea7124eddd4e287917be21bd0a0b7cb946816ad08c2a77943177a54569d87d0L443-L447

Reply:
yup, saw that as well. Great stuff.

Reply:
Good catch too. You had to look at the deleted file to find that.